internal/prober/prober_test.go

```diff
@@ -235,38 +211,43 @@ func runProber(p *Prober, d time.Duration) {
 	}
 }
 
-func checkProbeStatus(t *testing.T, ps probeStatus, successCount int, errCount int) {
-	g := NewWithT(t)
-	if successCount == 0 {
-		g.Expect(ps.successCount).To(Equal(successCount))
-	} else {
-		g.Expect(ps.successCount).To(BeNumerically(">=", successCount))
-	}
-	if errCount == 0 {
-		g.Expect(ps.errorCount).To(Equal(errCount))
-	} else {
-		g.Expect(ps.errorCount).To(BeNumerically(">=", errCount))
-	}
+func createMockInterfaces(t *testing.T) {
+	ctrl = gomock.NewController(t)
+	mockScaler = mockscaler.NewMockScaler(ctrl)
+	mockShootClientCreator = mockprober.NewMockShootClientCreator(ctrl)
+	mockKubernetesInterface = mockinterface.NewMockInterface(ctrl)
+	mockCoordinationV1Interface = mockcoordinationv1.NewMockCoordinationV1Interface(ctrl)
+	mockLeaseInterface = mockcoordinationv1.NewMockLeaseInterface(ctrl)
+	mockDiscoveryInterface = mockdiscovery.NewMockDiscoveryInterface(ctrl)
 }
 
-func setupProberTest(t *testing.T) {
-	ctrl = gomock.NewController(t)
-	mds = mockscaler.NewMockScaler(ctrl)
-	msc = mockprober.NewMockShootClientCreator(ctrl)
-	clientBuilder = fake.NewClientBuilder()
-	fakeClient = clientBuilder.Build()
-	mki = mockinterface.NewMockInterface(ctrl)
-	mdi = mockdiscovery.NewMockDiscoveryInterface(ctrl)
+func initializeMockInterfaces(entry probeSimulatorEntry) {
```

Comment:
can you rename this to `configureMocks` instead? 
If you decide to create these mocks for every test case then maybe you can club creation and configuration into a single function for a `ProberTestCase` and then call that function `createAndConfigureMocks`